### PR TITLE
net/context: Lock/unlock mutex on new send/sendto functions

### PR DESCRIPTION
**Pull request by tbursztyka**
This was forgotten modification as commit 93e5181fbdf came in master
though commit 0d519f7bcf7 was already written.

Thus fixing the missing lock/unlock.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

### CODE REVIEW

**Review by pfalcon — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Comment by tbursztyka** [comment]
weird CI failure... runs well on my host.

**Comment by tbursztyka** [comment]
recheck

File: subsys/net/ip/net_context.c [hunk 1]
```diff
@@ -977,7 +977,6 @@ int net_context_accept(struct net_context *context,
 
 	if (net_context_get_ip_proto(context) == IPPROTO_TCP) {
 		ret = net_tcp_accept(context, cb, user_data);
-		goto unlock;
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: I would not say this is useless. It tries to protect a case where someone puts code before the unlock: label and forgets to add the goto.

File: subsys/net/ip/net_context.c [hunk 2]
```diff
@@ -1673,7 +1672,6 @@ int net_context_recv(struct net_context *context,
 
 		if (ret == -EAGAIN) {
 			ret = -ETIMEDOUT;
-			goto unlock;
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: same here

**Review by jukkar — CHANGES_REQUESTED** [review]
I would like to keep the `goto unlock` even if they look useless now.

**Review by jukkar — APPROVED** [review]

**Pull request merged**